lib/ansible/plugins/shell/__init__.py

```diff
@@ -211,7 +211,7 @@ def build_module_command(self, env_string, shebang, cmd, arg_path=None):
             arg_path,
         ]
 
-        return f'{env_string}%s' % self.join(cps for cp in cmd_parts if cp and (cps := cp.strip()))
+        return f'{env_string}{self.join(cps for cp in cmd_parts if cp and (cps := cp.strip()))}'
```

Comment:
This highly expression is fairly difficult to consume. Mind splitting it?
```suggestion
        cleaned_up_cmd = self.join(
            stripped_cmd_part for raw_cmd_part in cmd_parts
            if raw_cmd_part and (stripped_cmd_part := raw_cmd_part.strip())
        )
        return ''.join((env_string, cleaned_up_cmd))
```